### PR TITLE
Added Kitchen and Travis [WIP]

### PR DESCRIPTION
**Pull request by Martin819**

### CODE REVIEW

**Comment by cloudlab-autobuild** [comment]
Can one of the admins verify this patch?

File: docker/client.sls [hunk 1]
```diff
@@ -10,8 +10,13 @@ include:
   - docker.client.service
   {%- endif %}
 
+{%- if not grains.get('oscodename', 'trusty') %}
```
Comment (fpytloun) [inline docker/client.sls]: WTF, why?

File: docker/client/compose.sls [hunk 2]
```diff
@@ -27,14 +30,15 @@ docker_compose_wrapper:
     - source: salt://docker/files/docker-compose
     - template: jinja
     - defaults:
-        image: {{ client.compose.source.image|default('docker/compose') }}
+        image: {{ client.compose.source.image|default('docker-compose') }}
```
Comment (fpytloun) [inline docker/client/compose.sls]: This is something new? I thought docker/compose is working fine.

File: docker/client/container.sls [hunk 3]
```diff
@@ -6,12 +6,34 @@ include:
...
     - require:
-      - pkg: docker_python
+      - pkg: python-docker
+
+{%- else %}
+
+{{id}}_image:
+  dockerng.image_present:
+    - name: {{ container.image }}
+    - reload_modules: true
```
Comment (fpytloun) [inline docker/client/container.sls]: Why are you adding reload modules everywhere? That will just slow down execution imho.

File: docker/client/container.sls [hunk 4]
```diff
@@ -6,12 +6,34 @@ include:
 {%- for name, container in client.get('container', {}).iteritems() %}
   {%- set id = name %}
   {%- set required_containers = [] %}
+{#
+get_docker_py:
```
Comment (fpytloun) [inline docker/client/container.sls]: No way, there are salt resources to work with pip anyway you want to make it possible to choose install method (pkg, pip, etc.) and prefer pkg where possible (eg. see installation of docker-compose)

File: docker/client/container.sls [hunk 5]
```diff
@@ -6,12 +6,34 @@ include:
...
+  cmd.run:
+  - name: 'apt-get -y install python-pip --force-yes && pip install docker-py'
+  - reload_modules: true
+#}
+{%- if not grains.get('oscodename', 'trusty') %}
```
Comment (fpytloun) [inline docker/client/container.sls]: Nope. If some parts are not supported on trusty, just skip it or fix correctly

File: docker/host.sls [hunk 6]
```diff
@@ -1,8 +1,22 @@
 {% from "docker/map.jinja" import host with context %}
 {%- if host.enabled %}
 
+sources_list:
+  cmd.run:
+  - name: 'apt-get -y install apt-transport-https --force-yes && apt-get -y install software-properties-common --force-yes && curl -fsSL https://apt.dockerproject.org/gpg | sudo apt-key add - && apt-key fingerprint 58118E89F3A912897C070ADBF76221572C52609D && add-apt-repository "deb https://apt.dockerproject.org/repo/ ubuntu-{{ host.oscodename }} main"'
```
Comment (fpytloun) [inline docker/host.sls]: Nooo way, these things are done by linux formula. You should never touch apt configuration from other formulas.

**Review by fpytloun — CHANGES_REQUESTED** [review]
See inline comments

**Pull request closed without merging**